app/src/test/java/com/teamagam/gimelgimel/app/model/ViewsModels/messages/MessagesViewModelTest.java

```diff
@@ -0,0 +1,185 @@
+
+        mDataChangedObserver = mock(DataChangedObserver.class);
+        mMessageViewModel.setObserver(mDataChangedObserver);
+    }
+
+
+    @Test
+    public void selectMessage_shouldFireDataChanged() throws Exception {
+        //Arrange
+        Message message = createMessage();
+        DisplayMessage displayMessage = new DisplayMessage(message);
+
+        //Act
+        mMessageViewModel.select(displayMessage);
+
+        //Assert
+        verify(mDataChangedObserver, atLeast(1)).onDataChanged();
+    }
+
+
+
+    @Test
+    public void onMessageModelChanged_shouldFireDataChanged() throws Exception {
+        //Arrange
+        mMessagesModel.add(createMessage());
+
+        //Assert
+        verify(mDataChangedObserver).onDataChanged();
+    }
+
+    @Test
+    public void onSelectedModelChanged_shouldFireDataChanged() throws Exception {
+        //Arrange
+        mSelectedMessageModel.select(createMessage());
+
+        //Assert
+        verify(mDataChangedObserver).onDataChanged();
+    }
+
+    @Test
+    public void onMessagesReadStatusModelChanged_shouldFireDataChanged() throws Exception {
+        //Arrange
+        mMessagesReadStatusModel.markAsRead(createMessage());
+
+        //Assert
+        verify(mDataChangedObserver).onDataChanged();
+
+    }
+
+    @Test
+    public void getDisplayedMessagesRandomAccessor_shouldReturnRandomAccessorOfAppropriateSize() throws Exception {
+        //Arrange
+        mMessagesModel.add(createMessage());
+        mMessagesModel.add(createMessage());
+        mMessagesModel.add(createMessage());
+
+        //Act
+        MessagesViewModel.DisplayedMessagesRandomAccessor res = mMessageViewModel.getDisplayedMessagesRandomAccessor();
+
+        //Assert
+        assertThat(res.size(), equalTo(3));
+    }
+
+    @Test
+    public void getDisplayedMessageRandomAccessor_shouldReturnMessagesInRightOrder() throws Exception {
+        //Arrange
+        Message m1 = createMessage();
+        Message m2 = createMessage();
+        Message m3 = createMessage();
+
+        mMessagesModel.add(m1);
+        mMessagesModel.add(m2);
+        mMessagesModel.add(m3);
+
+        //Act
+        MessagesViewModel.DisplayedMessagesRandomAccessor res = mMessageViewModel.getDisplayedMessagesRandomAccessor();
+
+        //Assert
+        assertThat(res.get(0).getMessage(), sameInstance(m1));
+        assertThat(res.get(1).getMessage(), sameInstance(m2));
+        assertThat(res.get(2).getMessage(), sameInstance(m3));
+    }
+
+    @Test
+    public void getDisplayedMessageRandomAccessorWithNoSelectedMessage_shouldReturnFirstMessageAsSelected() throws Exception {
+        //Arrange
+        mMessagesModel.add(createMessage());
+        mMessagesModel.add(createMessage());
+
+        //Act
+        MessagesViewModel.DisplayedMessagesRandomAccessor res = mMessageViewModel.getDisplayedMessagesRandomAccessor();
+
+        //Assert
+        assertThat(res.get(0).isSelected(), is(true));
```

Comment:
what should happen if I do the followin test:
        assertThat(res.get(1).isSelected(), is(true));
        assertThat(res.get(0).isSelected(), is(false));
I think this is OK